src/test/java/io/vertx/test/core/BlockedThreadCheckerTest.java

```diff
@@ -110,4 +112,82 @@ public void start() throws InterruptedException {
     await();
     blockedThreadWarning.expectMessage("vert.x-worker-thread", maxWorkerExecuteTime);
   }
+
+
+  @Test
+  public void testBlockCheckExceptionTimeLimitWithTimeUnit() throws Exception {
```

Comment:
can you refactor the tests to avoid code duplication with testBlockCheckExceptionTimeLimit, same for other methods